FaunaDB.Client/Reflection/Encoder.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FaunaDB.Query;
+using FaunaDB.Types;
+using FaunaDB.Attributes;
+
+namespace FaunaDB.Reflection
+{
+    public static class Encoder
+    {
+        public static Expr EncodeObject(object obj) =>
```

Comment:
`EncodeObject` gives the impression that I can't encode anything else, like lists for example. Is that correct?